### PR TITLE
feat: add simplfied associated type syntax

### LINKED ISSUE #6196: Add new associated type syntax
Without explicit parameter

### PR DESCRIPTION
**Pull request by mlutze**
fixes #6196

### CODE REVIEW

File: main/test/flix/Test.Assoc.Type.flix [hunk 1]
```diff
@@ -16,4 +16,20 @@ mod Test.Assoc.Type {
         pub def head(x: String): Option[Char] = if (String.isEmpty(x)) None else Some(String.charAt(0, x))
     }
 
+    class Collection2[t] {
+        type Elm: Type
```
Comment (magnus-madsen) [inline main/test/flix/Test.Assoc.Type.flix:20]: Can I elide the kind also, and have it default to Type?

File: main/test/flix/Test.Assoc.Type.flix [hunk 2]
```diff
@@ -16,4 +16,20 @@ mod Test.Assoc.Type {
         pub def head(x: String): Option[Char] = if (String.isEmpty(x)) None else Some(String.charAt(0, x))
     }
 
+    class Collection2[t] {
+        type Elm
```
Comment (magnus-madsen) [inline main/test/flix/Test.Assoc.Type.flix:20]: Maybe we can have an intermediate test where the kind is present? :) So we have 3 tests.

File: main/test/flix/Test.Assoc.Type.flix [hunk 3]
```diff
@@ -16,4 +16,20 @@ mod Test.Assoc.Type {
...
+
+        pub def isEmpty(x: t): Bool
+        pub def contains(x: t, e: Collection2.Elm[t]): Bool
```
Comment (magnus-madsen) [inline main/test/flix/Test.Assoc.Type.flix:23]: Later I assume we will put `Elm` into the relevant scope?

Comment (mlutze) [reply main/test/flix/Test.Assoc.Type.flix:23]: Yes. This is related to another issue where we have to qualify other signatures.

**Review by magnus-madsen — APPROVED** [review]

**Comment by magnus-madsen** [comment]
Excellent. I think this syntax is much preferable (at least for simple cases).

File: main/test/flix/Test.Assoc.Type.flix (hunk 3 above)
Comment (magnus-madsen) [reply main/test/flix/Test.Assoc.Type.flix:23]: Is this fixable or requires a lot of work?

Comment (mlutze) [reply main/test/flix/Test.Assoc.Type.flix:23]: Fixable but will be a breaking change

**Pull request merged**